packages/eslint-plugin/src/rules/max-params.ts

```diff
@@ -0,0 +1,97 @@
+import { AST_NODE_TYPES, type TSESTree } from '@typescript-eslint/utils';
+import type { JSONSchema4 } from '@typescript-eslint/utils/json-schema';
+
+import * as util from '../util';
+import { getESLintCoreRule } from '../util/getESLintCoreRule';
+
+type FunctionLike =
+  | TSESTree.FunctionDeclaration
+  | TSESTree.FunctionExpression
+  | TSESTree.ArrowFunctionExpression;
+
+type FunctionRuleListener<T extends FunctionLike> = (node: T) => void;
+
+const baseRule = getESLintCoreRule('max-params');
+
+export type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+const schema = [
+  {
+    type: 'object',
+    properties: {
+      maximum: {
+        type: 'integer',
+        minimum: 0,
+      },
+      max: {
+        type: 'integer',
+        minimum: 0,
+      },
+      countVoidThis: {
+        type: 'boolean',
+      },
+    },
+    additionalProperties: false,
+  },
+] as JSONSchema4[];
```

Comment:
for some reason, I can't get `deepMerge` to work properly here. 

I've tried adding these lines to add support for array overrides, but I'm not sure why it doesn't work:
```diff
 if (isObjectNotArray(firstValue) && isObjectNotArray(secondValue)) {
   // object type
   acc[key] = deepMerge(firstValue, secondValue);
+ } else if (Array.isArray(firstValue)) {
+   // array type
+   acc[key] = Object.values(
+     deepMerge({ ...firstValue }, { ...(secondValue as ObjectLike) }),
+   );
  } else {
   // value type
   acc[key] = secondValue;
 }
```

seems to work fine when isolated though:
https://tsplay.dev/wX1GLN
___
Anyway... I should've probably still hard-coded the values here since I don't support passing only a number like the base rule does (maybe I should?)

Reply:
Extended rules generally directly go off `context.options`, is that what you're looking for?

https://github.com/typescript-eslint/typescript-eslint/blob/ca1abb5cdaecd8f039a807b47db2a642e14fbc1e/packages/eslint-plugin/src/rules/brace-style.ts#L30-L32

Reply:
Nope, I'm talking about `baseRule.meta.schema`.
I had some issues when trying to add an option to the base schema, the deepMerge function seems to break it.
Is it OK to keep it hard-coded instead of extending the original schema in runtime?

Reply:
Ah got it - yeah this is fine. Base rules don't change often and many others already hard-code it (e.g. `dot-notation`). Thanks for asking!